nativelib/_x32/src/main/scala/scala/scalanative/runtime/Platform.scala

```diff
@@ -0,0 +1,21 @@
+package scala.scalanative
+package runtime
+
+import scala.scalanative.native.{CFunctionPtr2, CString}
+
+object Platform {
+  def isMac(): Boolean                 = ExternPlatform.isMac()
+  def isWindows(): Boolean             = ExternPlatform.isWindows()
+  def windowsGetUserLang(): CString    = ExternPlatform.windowsGetUserLang()
+  def windowsGetUserCountry(): CString = ExternPlatform.windowsGetUserCountry()
+  def littleEndian(): Boolean          = ExternPlatform.littleEndian()
+  def setOSProps(addProp: CFunctionPtr2[CString, CString, Unit]): Unit =
+    ExternPlatform.setOSProps(addProp)
+
+  @inline final def cross3264[A, B](thirtyTwo: => A,
```

Comment:
The reason I defined the method like this was to handle different types across 32/64 bit platforms, such as returning an Int vs Long if a `CLong` is expected. Is there a way to allow this to be optimized without losing the type safety?

Reply:
Indeed, there are benefits in terms of type safety. But from a pragmatic point of view, you can't have afford having the default implementation of equals and hashCode to allocate closures. A compromise where those kind of cases use a `if (isCross32) ... else ...` and the others (non-performance-sensitive) parts use the type safe version would be fine with me.

Reply:
That makes sense. Is there a reason there would still be closures allocated even after `cross3264` is inlined?

Reply:
LLVM's IR is around C level's of abstraction (structs, arrays and pointers). So even after inlining of the `cross3264` it would still keep the allocation and virtual dispatch as it doesn't really know how to deal with those.